src/functionaltests/java/com/ericsson/gerrit/plugins/eiffel/linking/GerritMockTest.java

```diff
@@ -0,0 +1,187 @@
+        gerritMock.createBranch(branch);
+        GitCommit base = gerritMock.getHead(branch);
+
+        String changeId = gerritMock.createNewChange(user1, branch);
+        GitCommit update = gerritMock.getCommit(changeId);
+        assertEquals("Not correct parrent", base.sha, update.parentSha);
+
+        GitCommit newPatchSet = gerritMock.createNewPatchSet(changeId);
+        assertEquals("Parent changed when creating new patchset", base.sha, newPatchSet.parentSha);
+
+        GitCommit merged = gerritMock.submit(changeId);
+        assertEquals("The submitted commit does not have correct parent", base.sha,
+                merged.parentSha);
+        assertEquals("The branch is not updated correctly", merged.sha,
+                gerritMock.getHead(branch).sha);
+    }
+
+    @Test
+    public void oneUserTwoBranches() {
+        GerritMock gerritMock = new GerritMock();
+        String branch1 = "branch1";
+        String branch2 = "branch2";
+        String user1 = "user1";
+        gerritMock.createBranch(branch1);
+        gerritMock.createBranch(branch2);
+        GitCommit base = gerritMock.getHead(branch1);
+        GitCommit base2 = gerritMock.getHead(branch2);
+
+        String changeId = gerritMock.createNewChange(user1, branch1);
+        assertEquals("Lookup did not return correct changId", changeId,
+                gerritMock.getChangeId(user1, branch1));
+
+        GitCommit update = gerritMock.getCommit(changeId);
+        assertEquals("Not correct parrent", base.sha, update.parentSha);
+
+        String changeId2 = gerritMock.createNewChange(user1, branch2);
+        assertEquals("Lookup did not return correct changId", changeId2,
+                gerritMock.getChangeId(user1, branch2));
+
+        GitCommit update2 = gerritMock.getCommit(changeId2);
+        assertEquals("Not correct parrent", base2.sha, update2.parentSha);
+
+        GitCommit newPatchSet = gerritMock.createNewPatchSet(changeId);
+        assertEquals("Parent changed when creating new patchset", base.sha, newPatchSet.parentSha);
+
+        GitCommit newPatchSet2 = gerritMock.createNewPatchSet(changeId2);
+        assertEquals("Parent changed when creating new patchset", base2.sha,
+                newPatchSet2.parentSha);
+
+        GitCommit merged = gerritMock.submit(changeId);
+        assertEquals("The submitted commit does not have correct parent", base.sha,
+                merged.parentSha);
+        assertEquals("The branch is not updated correctly", merged.sha,
+                gerritMock.getHead(branch1).sha);
+
+        GitCommit merged2 = gerritMock.submit(changeId2);
+        assertEquals("The submitted commit does not have correct parent", base2.sha,
+                merged2.parentSha);
+        assertEquals("The branch is not updated correctly", merged2.sha,
+                gerritMock.getHead(branch2).sha);
+    }
+
+    @Test
+    public void twoUsers() {
+        GerritMock gerritMock = new GerritMock();
```

Comment:
can we use the gerritmock in  @Before part and use that, as it is common in most of the cases.

Reply:
It would only save one line of code and makes it slight harder to read

Reply:
I agree with Santosh, and the branches could be static variables as well, doing this would save 2-3 lines for each test. There is also an empty line at line 171.

Reply:
I have changed to constants in 78d15fa I feel it improves readablity but I don't want to move the new gerrit mock into before due to I feel it decreases readability.